OneCommunityGlobal/HGNRest
# Renan luiz fix x hours in x weeks streak badges

# Description
![image](https://github.com/OneCommunityGlobal/HGNRest/assets/91035340/691d2a2c-bdd5-43d0-99b4-850c32390826)

## Related PRS (if any):
To test this backend PR you need to check the latest frontend

## Main changes explained:
it was included a simple logic in checkXHoursXWeeks function
…

## How to test:
check into current branch
do npm install and ... to run this PR locally
Clear site data/cache
you need to add to run awardNewBadges every time
you need to take off awardNewBadges function inside condicional
you need to enter in the mongodb compass
you need to check userProfile folder
you need to add the hours you would like to get the badge streak in savedTangibleHours array
you need to monitor if in BadgeCollections appear only one badge streak
see the tutorials how to test below:

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HGNRest/assets/91035340/f53a7e07-22ec-4371-9dce-a36978406e17


https://github.com/OneCommunityGlobal/HGNRest/assets/91035340/2b618038-2b44-4b33-ac45-4885146413a8


https://github.com/OneCommunityGlobal/HGNRest/assets/91035340/369c04f6-d675-429b-80f5-335586aa5993


## Note:

Include the information the reviewers need to know.

if the badge didn't appear in the badgeCollection, probably the badge don't exist
if you have some difficults to test the back-end in nodemon, please watch this video: https://www.youtube.com/watch?v=GYR6GSAVC_k or this video https://www.dropbox.com/scl/fi/223fqqbzultw6chbl9a5p/Backend-testing-guide.mkv?rlkey=ygsr5w91lf76yxdha9n83favs&dl=0
Some questions please send me a dm in slack
